lib/oban/engines/basic.ex

```diff
@@ -182,10 +182,24 @@ defmodule Oban.Engines.Basic do
   end
 
   @impl Engine
-  def cancel_job(%Config{} = conf, %Job{id: id}) do
-    query = where(Job, [j], j.id == ^id)
+  def cancel_job(%Config{} = conf, %Job{} = job) do
+    updates = [set: [state: "cancelled", cancelled_at: utc_now()]]
+
+    updates =
+      if is_map(job.unsaved_error) do
+        error = %{attempt: job.attempt, at: utc_now(), error: format_blamed(job.unsaved_error)}
+
+        Keyword.put(updates, :push, errors: error)
```

Comment:
When I changed from returning `{:discard, reason}` to `{:cancel, reason}`
The `reason` was not recorded in errors any more

Because I'm using oban_pro SmartEngine, so I think oban_pro 0.12 might be missing this logic